jerry-ext/debugger/debugger-tcp.c

```diff
@@ -19,28 +19,102 @@
 
 #ifdef JERRY_DEBUGGER
 
-#include <arpa/inet.h>
 #include <errno.h>
+
+#if defined (WIN32)
+#include <BaseTsd.h>
+typedef SSIZE_T ssize_t;
+#include <WS2tcpip.h>
+#include <winsock2.h>
+
+/* On Windows the WSAEWOULDBLOCK value can be returned for non-blocking operations */
+#define JERRYX_EWOULDBLOCK WSAEWOULDBLOCK
+
+/* On Windows the invalid socket's value of INVALID_SOCKET */
+#define JERRYX_SOCKET_INVALID INVALID_SOCKET
+
+/* On Windows sockets have a SOCKET typedef */
+typedef SOCKET jerryx_socket;
+
+#else /* !WIN32 */
+
+#include <arpa/inet.h>
 #include <fcntl.h>
-#include <unistd.h>
 #include <sys/socket.h>
+#include <unistd.h>
+
+/* On *nix the EWOULDBLOCK errno value can be returned for non-blocking operations */
+#define JERRYX_EWOULDBLOCK EWOULDBLOCK
+
+/* On *nix the invalid socket has a value of -1 */
+#define JERRYX_SOCKET_INVALID (-1)
+
+/* On *nix the sockets are integer identifiers */
+typedef int jerryx_socket;
+#endif /* WIN32 */
 
 /**
  * Implementation of transport over tcp/ip.
  */
 typedef struct
 {
   jerry_debugger_transport_header_t header; /**< transport header */
-  int tcp_socket; /**< tcp socket */
+  jerryx_socket tcp_socket; /**< tcp socket */
 } jerryx_debugger_transport_tcp_t;
 
+/**
+ * Get the network error value.
+ *
+ * On Windows this returns the result of the `WSAGetLastError ()` call and
+ * on any other system the `errno` value.
+ *
+ *
+ * @return last error value.
+ */
+static inline int
+jerryx_debugger_tcp_get_errno (void)
+{
+#ifdef WIN32
+  return WSAGetLastError ();
+#else /* !WIN32 */
+  return errno;
+#endif /* WIN32 */
+} /* jerryx_debugger_tcp_get_errno */
+
+/**
+ * Correctly close a single socket.
+ */
+static inline void
+jerryx_debugger_tcp_close_socket (jerryx_socket socket_id) /**< socket to close */
+{
+#ifdef WIN32
+  closesocket (socket_id);
+#else /* !WIN32 */
+  close (socket_id);
+#endif /* WIN32 */
+} /* jerryx_debugger_tcp_close_socket */
+
 /**
  * Log tcp error message.
  */
 static void
-jerryx_debugger_tcp_log_error (int err_val)
+jerryx_debugger_tcp_log_error (const char *prefix, /**< message prefix */
```

Comment:
minor: is the `prefix` really necessary? Are not all errors TCP-related so that it shouldn't be hardcoded? This is the TCP transport implementation after all.

Reply:
Initially I wanted to use the simple `TCP error` hard-coded prefix, but in this variant I've followed the original `Error/TCP Error` prefixes. I'm happy to change this to the "TCP Error" prefix.

Reply:
I see. Well, then, it's no big deal. I vote for a single prefix but whatever majority votes for should prevail.